src/test/kotlin/org/rust/ide/inspections/RsMatchCheckInspectionTest.kt

```diff
@@ -1227,4 +1227,39 @@ class RsMatchCheckInspectionTest : RsInspectionsTestBase(RsMatchCheckInspection:
             }
         }
     """)
+
+    fun `test non-exhaustive enum match in same crate`() = checkByText("""
+        #[non_exhaustive]
+        enum Error {
+            Variant
+        }
+
+        fn main() {
+            let error = Error::Variant;
+
+            match error {
+                Error::Variant => println!("Variant")
+            }
+        }
+    """)
+
+    fun `test non-exhaustive enum match in different crate`() = checkByFileTree("""
```

Comment:
This test currently fails. It doesn't run the inspection code, so I probably don't have it set up correctly.

Reply:
Hey, let's replace it with the new code below:

```kotlin
    @ProjectDescriptor(WithDependencyRustProjectDescriptor::class)
    fun `test non-exhaustive enum match in different crate`() = checkByFileTree("""
    //- main.rs
        extern crate dep_lib_target;
        use dep_lib_target::Error;

        fn main() {
            let error = Error::Variant;

            <error descr="Match must be exhaustive [E0004]">match/*caret*/</error> error {
                Error::Variant => println!("Variant")
            }
        }
    //- dep-lib/lib.rs
        #[non_exhaustive]
        pub enum Error {
            Variant
        }
    """)
```